packages/remix-react/routes.tsx

```diff
@@ -249,16 +249,34 @@ export function createClientRoutes(
   return (routesByParentId[parentId] || []).map((route) => {
     let routeModule = routeModulesCache[route.id];
 
-    async function fetchServerLoader(request: Request) {
+    async function fetchServerLoader(
+      request: Request,
+      unwrap: boolean,
+      singleFetch: unknown
+    ) {
       if (!route.hasLoader) return null;
-      return fetchServerHandler(request, route);
```

Comment:
Instead of always hitting the server here, if single fetch is enabled we will have passed the `singleFetch` method down from `dataStrategy` so use that to ensure we make the right HTTP call to the server

Reply:
By proxying this all the way though and branching here as deep as possible - we avoid having to re-implement any of the logic around critical/lazy routes, revalidation, client data etc. in data
strategy and get to leverage all of the existing battle-tested flows.